lib/modules/dosomething/dosomething_home/dosomething_home.module

```diff
@@ -88,7 +90,7 @@ function dosomething_home_preprocess_node(&$vars) {
       $tiles['image'] = '<img src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs=" data-src="' . $tiles['image_url'] . '">';
 
       $link_language = dosomething_global_get_language($user, $node, FALSE);
-      $tiles['link'] = dosomething_global_node_url($nid, $link_language);
+      $tiles['link'] = dosomething_global_url('node/' . $nid, ['language' => $link_language, 'query' => ['source' => $source]]);
```

Comment:
one final nit picky thing, but if you're going to use a variable for `node/ . $nid` use it for both, otherwise scrap the temp var